internal/pkg/template/input/input.go

```diff
@@ -82,11 +82,21 @@ type Input struct {
 	Rules       Rules       `json:"rules,omitempty" validate:"omitempty,template-input-rules"`
 	If          If          `json:"showIf,omitempty" validate:"omitempty,template-input-if"`
 	Options     Options     `json:"options,omitempty" validate:"template-input-options"`
-	ComponentId string      `json:"componentId,omitempty"`
+	ComponentId string      `json:"componentId,omitempty" validate:"required_if=Kind oauth"`
 }
 
 // ValidateUserInput validates input from the template user using Input.Rules.
 func (i Input) ValidateUserInput(userInput interface{}) error {
+	if i.Kind == KindOAuth {
+		switch v := userInput.(type) {
+		case map[string]interface{}:
+			if len(v) == 0 {
+				return fmt.Errorf("oauth must not be empty")
+			}
+		default:
+			// handle unknown type
+		}
+	}
```

Comment:
Tohle trochu krkolomně říká že pokud `kind` je `oauth` a `userInput` má ten správný typ, tak se zkontroluje že není prázdný. Testuje se to v `internal/pkg/template/input/input_test.go`. 

(Ten type switch je tam proto, že by tam potenciálně někdo mohl poslat nějakou hodnotu typu nad kterým se nedá volat `len()` funkce a to by potom vyhazovalo panic error.)